backends/edk2-compat/edk2-svc-generate.c

```diff
@@ -232,200 +206,148 @@ int performGenerateCommand(int argc,char* argv[])
+                rc = ARG_PARSE_FAIL;
+                break;
             }
-			// set public key signer
-			else if(!strcmp(argv[i], "-c")) {	
-				if (i + 1 >= argc || argv[i + 1][0] == '-') {
-					prlog(PR_ERR, "ERROR: Incorrect value for public key flag, use 'c <cert>', see usage...\n");
-					rc = ARG_PARSE_FAIL;
-					goto out;
-				}
-				else {
-					i++;
-					args->signCertCount++;
-					args->signCerts = realloc(args->signCerts, args->signCertCount * sizeof(char*));
-					args->signCerts[args->signCertCount - 1] = argv[i];
-				}
-			}
-			// set input file
-			else if (!strcmp(argv[i], "-i")) {
-				if (i + 1 >= argc || argv[i + 1][0] == '-') {
-					prlog(PR_ERR, "ERROR: Incorrect flag '-i', see usage...\n");
-					rc = ARG_PARSE_FAIL;
-					goto out;
-				}
-				else {
-					i++;
-					args->inFile = argv[i];
-				}
-			}
-			// set output file 
-			else if (!strcmp(argv[i], "-o")) {
-				if (i + 1 >= argc || argv[i + 1][0] == '-') {
-					prlog(PR_ERR, "ERROR: Incorrect flag '-o', see usage...\n");
-					rc = ARG_PARSE_FAIL;
-					goto out;
-				}
-				else {
-					i++;
-					args->outFile = argv[i];
-				}
+            args->alreadySignedFlag = 0;
+			args->signKeyCount++;
+			args->signKeys = realloc(args->signKeys, args->signKeyCount * sizeof(char*));
+			args->signKeys[args->signKeyCount - 1] = arg;
+			break;
+		case 'c':
+			args->signCertCount++;
+			args->signCerts = realloc(args->signCerts, args->signCertCount * sizeof(char*));
+			args->signCerts[args->signCertCount - 1] = arg;
+			break;
+		case 'f':
+			args->inpValid = 1;
+			break;
+		case 'v':
+			verbose = PR_DEBUG;
+			break;
+		case 's':
+			//if already storing private keys, then don't allow for signed data
+            if (args->alreadySignedFlag == 0){
+                prlog(PR_ERR, "ERROR: Cannot have both signed data files and private keys for signing");
+                rc = ARG_PARSE_FAIL;
+                break;
+            }
+            args->alreadySignedFlag = 1;
+            args->signKeyCount++;
+            args->signKeys = realloc(args->signKeys, args->signKeyCount * sizeof(char*));
+            args->signKeys[args->signKeyCount - 1] = arg;
+			break;
+		case 'i':
+			args->inFile = arg;
+			break;
+		case 'o':
+			args->outFile = arg;
+			break;
+		case 'h':
+			args->hashAlg = arg;
+			break;
+		case 'n':
+			args->varName = arg;
+			break;
+		case 't':
+			args->time = calloc(1, sizeof(*args->time));
+			if (!args->time) {
+				prlog(PR_ERR, "ERROR: failed to allocate memory\n");
+				rc = ALLOC_FAIL;
+				break;
 			}
-			// set variable name
-			else if (!strcmp(argv[i], "-n")) {
-				if (i + 1 >= argc || argv[i + 1][0] == '-') {
-					prlog(PR_ERR, "ERROR: Incorrect flag '-n', see usage...\n");
-					rc = ARG_PARSE_FAIL;
-					goto out;
-				}
-				else {
-					i++;
-					args->varName = argv[i];
-				}
+			//make sure -t flag has two strings <date> <time>
```

Comment:
I'm pretty uncomfortable with this as an CLI design decision, I think -t should take 1 string "<date> <time>". But @erichte-ibm can weigh in.

Reply:
I will wait for @erichte-ibm  to chime in. I have no strong opinions on this but would prefer to get things moving so we can make RHEL 8.5 .